src/main/native/org_cryptomator_windows_keychain_WinHello_Native.cpp

```diff
@@ -0,0 +1,206 @@
+  return array;
+}
+
+IBuffer vectorToIBuffer(const std::vector<uint8_t>& vec) {
+  DataWriter writer;
+  writer.WriteBytes(array_view<const uint8_t>(vec));
+  return writer.DetachBuffer();
+}
+
+std::vector<uint8_t> iBufferToVector(const IBuffer& buffer) {
+  auto reader = DataReader::FromBuffer(buffer);
+  if (!buffer || reader.UnconsumedBufferLength() == 0) {
+    return {};
+  }
+  std::vector<uint8_t> result(reader.UnconsumedBufferLength());
+  reader.ReadBytes(array_view<uint8_t>(result));
+  return result;
+}
+
+// Bring Windows Hello pop-up to the front
+void queueSecurityPromptFocus(int delay = 500) {
+    std::thread([delay]() {
+        while (g_promptFocusCount <= 3) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(delay));
+
+            auto hWnd = ::FindWindowA("Credential Dialog Xaml Host", nullptr);
+
+            if (hWnd) {
+                ::SetForegroundWindow(hWnd);
+                g_promptFocusCount = 0;
+                break;
+            } else if (++g_promptFocusCount > 3) {
+                g_promptFocusCount = 0;
+                break;
+            }
+        }
+    }).detach();
+}
+
+void InitializeWindowsRuntime() {
+    std::call_once(runtimeInitFlag, []() {
+        winrt::init_apartment();
+    });
+}
+
+bool deriveEncryptionKey(const std::vector<uint8_t>& challenge, std::vector<uint8_t>& key){
+  auto challengeBuffer = CryptographicBuffer::CreateFromByteArray(
+        array_view<const uint8_t>(challenge.data(), challenge.size()));
+
+  try {
+    // The first time this is used a key-pair will be generated using the common name
+    auto result = KeyCredentialManager::RequestCreateAsync(s_winHelloKeyName,
+          KeyCredentialCreationOption::FailIfExists).get();
+
+    if (result.Status() == KeyCredentialStatus::CredentialAlreadyExists) {
+      result = KeyCredentialManager::OpenAsync(s_winHelloKeyName).get();
+    } else if (result.Status() != KeyCredentialStatus::Success) {
+      std::cout << "Failed to retrieve Windows Hello credential." << std::endl;
+      return false;
+    }
+
+    const auto signature = result.Credential().RequestSignAsync(challengeBuffer).get();
+    if (signature.Status() != KeyCredentialStatus::Success) {
+      if (signature.Status() != KeyCredentialStatus::UserCanceled) {
+        std::cout << "Failed to sign challenge using Windows Hello." << std::endl;
+      }
+      return false;
+    }
+
+    // Use the SHA-256 hash of the challenge signature as the encryption key
+    const auto response = signature.Result();
+    HashAlgorithmProvider hashProvider = HashAlgorithmProvider::OpenAlgorithm(HashAlgorithmNames::Sha256());
+    IBuffer hashBuffer = hashProvider.HashData(response);
+    key = iBufferToVector(hashBuffer);
+    return true;
+
+  } catch (winrt::hresult_error const& ex) {
+    std::cout << winrt::to_string(ex.message()) << std::endl;
+    return false;
+  }
```

Comment:
Regarding Handle Exceptions More Precisely in deriveEncryptionKey: this is not necessary. The log messages are sufficient.

Regarding Use Secure Key Derivation Functions (KDFs): this. is done with 9ac7cb65323f6758a60ed883c9653b9a2d1b366c.

Regarding Check the Availability of Windows Hello Credential: this is not needed as the windows code guarantees to return a Credential that is not null. Otherwise an exception is thrown.